soak-tests-common-sql/src/main/java/com/hazelcast/jet/tests/common/sql/SqlResultProcessor.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.tests.common.sql;
+
+import com.hazelcast.sql.SqlResult;
+import com.hazelcast.sql.SqlService;
+import com.hazelcast.sql.SqlStatement;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+
+public class SqlResultProcessor {
+
+    private final String sqlString;
+    private final SqlService sqlService;
+
+    private final ExecutorService threadPool;
+
+    public SqlResultProcessor(String sqlString, SqlService sqlService, ExecutorService threadPool) {
+        this.sqlString = sqlString;
+        this.sqlService = sqlService;
+        this.threadPool = threadPool;
+    }
+
+    public Future<SqlResult> runQueryAsync() {
+        CompletableFuture<SqlResult> completableFuture = new CompletableFuture<>();
+        threadPool.submit(
+                () -> {
+                    SqlStatement statement = new SqlStatement(sqlString);
+                    try {
+                        SqlResult result = sqlService.execute(statement);
+                        completableFuture.complete(result);
+
+                    } catch (Throwable e) {
+                        e.printStackTrace();
+                        completableFuture.completeExceptionally(e);
+                    }
+                    return null;
+                }
+        );
+        return completableFuture;
+    }
+
+    public SqlResult awaitQueryExecutionWithTimeout(Future<SqlResult> sqlResultFuture, int timeoutSeconds) {
+        SqlResult sqlResult = null;
+
+        try {
+            sqlResult = sqlResultFuture.get(timeoutSeconds, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        } catch (ExecutionException e) {
```

Comment:
Why we need to `e.printStackTrace()` here? (it's printed in `AbstractSoakTest` and it should be rather part of exception). Instead of:
```
        } catch (ExecutionException e) {
            e.printStackTrace();
            throw new RuntimeException("SQL Query: " + sqlString + " has failed with reason " + e.getMessage());
        } catch (TimeoutException e) {
            e.printStackTrace();
            throw new RuntimeException("SQL Query: " + sqlString + " have not return value " + timeoutSeconds +
                    e.getMessage());
        }
```
you can do just:
```
        } catch (ExecutionException | TimeoutException e) {
            throw new RuntimeException("SQL Query: " + sqlString, e);
        } 
```

Reply:
Agreed simplified